nimiqclient/nimiq_client.py

```diff
@@ -0,0 +1,560 @@
+        When no parameter is given, it returns the current state.
+        When a value is given as parameter, it sets the current state to that value.
+        :param state: The state to be set.
+        :return: True if the client is mining, otherwise False.
+        """
+        params = []
+        if state != None:
+            params.append(state)
+        return self.call("mining", params)
+
+    def peerCount(self):
+        """
+        Returns number of peers currently connected to the client.
+        :return: Number of connected peers.
+        """
+        return self.call("peerCount", [])
+
+    def peerList(self):
+        """
+        Returns list of peers known to the client.
+        :return: The list of peers.
+        """
+        return self.call("peerList", [])
+
+    def peerState(self, address, command = None):
+        """
+        Returns the state of the peer.
+        When no command is given, it returns peer state.
+        When a value is given for command, it sets the peer state to that value.
+        :param address: The address of the peer.
+        :param command: The command to send.
+        :return: The current state of the peer.
+        """
+        params = []
+        params.append(address)
+        if command != None:
+            params.append(command)
+        return self.call("peerState", params)
+
+    def pool(self, address = None):
+        """
+        Returns or sets the mining pool.
+        When no parameter is given, it returns the current mining pool.
+        When a value is given as parameter, it sets the mining pool to that value.
+        :param address: The mining pool connection string ("url:port") or boolean to enable/disable pool mining.
+        :return: The mining pool connection string, or None if not enabled.
+        """
+        params = []
+        if address != None:
+            params.append(address)
+        return self.call("pool", params)
+
+    def poolConfirmedBalance(self):
+        """
+        Returns the confirmed mining pool balance.
+        :return: The confirmed mining pool balance (in smallest unit).
+        """
+        return self.call("poolConfirmedBalance", [])
+
+    def poolConnectionState(self):
+        """
+        Returns the connection state to mining pool.
+        :return: The mining pool connection state.
+        """
+        return self.call("poolConnectionState", [])
+
+    def sendRawTransaction(self, transaction):
+        """
+        Sends a signed message call transaction or a contract creation, if the data field contains code.
+        :param transaction: The hex encoded signed transaction
+        :return: The Hex-encoded transaction hash.
+        """
+        return self.call("sendRawTransaction", [transaction])
+
+    def sendTransaction(self, transaction):
+        """
+        Creates new message call transaction or a contract creation, if the data field contains code.
+        :param transaction: The hex encoded signed transaction
```

Comment:
Checking the tests, in line 189, the transaction object is JSON.

Reply:
Will be fixed in the doc for `sendTransaction`.

Reply:
The doc is fixed and now all tests use [predefined data structures](https://github.com/rraallvv/python-client/blob/master/test/test_nimiq_client.py#L181) instead of json objects.